applications/MappingApplication/custom_mappers/coupling_geometry_mapper.cpp

```diff
@@ -0,0 +1,332 @@
+                    rLocalMappingMatrix(i, j) = sf_values_slave(integration_point_itr, i)
+                        * sf_values_master(integration_point_itr, j)
+                        * det_jacobian[integration_point_itr];
+
+                    KRATOS_DEBUG_ERROR_IF(sf_values_master(integration_point_itr, i) < 0.0)
+                        << "SHAPE FUNCTIONS LESS THAN ZERO\n" << sf_values_master << std::endl;
+                    KRATOS_DEBUG_ERROR_IF(sf_values_slave(integration_point_itr, j) < 0.0)
+                        << "SHAPE FUNCTIONS LESS THAN ZERO\n" << sf_values_slave << std::endl;
+                }
+            }
+        }
+    }
+
+    for (IndexType i=0; i< sf_values_master.size2(); ++i) {
+        rOriginIds[i] = r_geometry_master[i].GetValue(INTERFACE_EQUATION_ID);
+    }
+    for (IndexType i=0; i< sf_values_slave.size2(); ++i) {
+        rDestinationIds[i] = r_geometry_slave[i].GetValue(INTERFACE_EQUATION_ID);
+    }
+}
+
+std::string CouplingGeometryLocalSystem::PairingInfo(const int EchoLevel) const
+{
+    //std::cout << "   >>> XXX : IS_PROJECTED_LOCAL_SYSTEM: " << mIsProjection << std::endl;
+    // KRATOS_DEBUG_ERROR_IF_NOT(mpNode) << "Members are not intitialized!" << std::endl;
+
+    // std::stringstream buffer;
+    // buffer << "CouplingGeometryLocalSystem based on " << mpNode->Info();
+    // if (EchoLevel > 1) { // TODO leave here?
+    //     buffer << " at Coodinates " << Coordinates()[0] << " | " << Coordinates()[1] << " | " << Coordinates()[2];
+    //     if (mPairingStatus == MapperLocalSystem::PairingStatus::Approximation) {
+    //         mpNode->SetValue(PAIRING_STATUS, 0);
+    //     } else {
+    //         mpNode->SetValue(PAIRING_STATUS, -1);
+    //     }
+    // }
+    // return buffer.str();
+    return "";
+}
+
+
+
+template<class TSparseSpace, class TDenseSpace>
+void CouplingGeometryMapper<TSparseSpace, TDenseSpace>::InitializeMapper()
+{
+    mpModeler = (ModelerFactory::Create(mMapperSettings["modeler_name"].GetString(), mrModel, mMapperSettings["modeler_parameters"]));
+    mpModeler->SetupGeometryModel();
+    mpModeler->PrepareGeometryModel();
+
+    // here use whatever ModelPart(s) was created by the Modeler
+    mpCouplingMP = &(mrModel.GetModelPart("coupling"));
+    mpCouplingInterfaceOrigin = mpCouplingMP->pGetSubModelPart("interface_origin");
+    mpCouplingInterfaceDestination = mpCouplingMP->pGetSubModelPart("interface_destination");
+
+    mpInterfaceVectorContainerOrigin = Kratos::make_unique<InterfaceVectorContainerType>(*mpCouplingInterfaceOrigin);
+    mpInterfaceVectorContainerDestination = Kratos::make_unique<InterfaceVectorContainerType>(*mpCouplingInterfaceDestination);
+
+
+    // TODO update
+    for (auto condition_itr = mpCouplingMP->ConditionsBegin();
+        condition_itr != mpCouplingMP->ConditionsEnd();
+        ++condition_itr)
+        condition_itr->GetGeometry().SetValue(IS_DUAL_MORTAR, mMapperSettings["dual_mortar"].GetBool());
```

Comment:
@philbucher @tteschemacher this is probably the dirtiest my code could get 😄 . We have to find another way to get the bool is_dual_mortar into the local mappers, does adding processinfo sound OK?

Reply:
there is no process info around, I adding it would be quite cumbersome

Before trying to find a better solution I have a general question. We basically have three versions of the `CouplingGeometryLocalSystem` right?
- projected mass matrix
- consistent slave matrix
- consistent slave matrix - dual mortar

Reply:
Yep just those three